model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1093,6 +1093,72 @@ message SideInput {
   FunctionSpec window_mapping_fn = 3;
 }
 
+message StandardArtifacts {
+  enum Types {
+    // A URN for artifacts stored in a local directory.
+    // payload: ArtifactFilePayload.
+    FILE     = 0 [(beam_urn) = "beam:artifact:file:v1"];
+
+    // A URN for artifacts embedded in ArtifactInformation proto.
+    // payload: EmbeddedFilePayload.
+    EMBEDDED = 1 [(beam_urn) = "beam:artifact:embedded:v1"];
+
+    // A URN for artifacts described by remote URLs.
+    // payload: a string for an artifact URL e.g. "https://.../foo.jar"
+    REMOTE     = 2 [(beam_urn) = "beam:artifact:remote:v1"];
+
+    // A URN for Python artifacts hosted on PYPI.
+    // payload: PypiPayload
+    PYPI     = 3 [(beam_urn) = "beam:artifact:pypi:v1"];
+
+    // A URN for Java artifacts hosted on Maven central.
+    // payload: MavenPayload
+    MAVEN    = 4 [(beam_urn) = "beam:artifact:maven:v1"];
+  }
+}
+
+message ArtifactFilePayload {
+  // A path to an artifact file on a local system.
+  string local_path = 1;
+
+  // A generated staged name (no path).
+  string staged_name = 2;
+}
+
+message EmbeddedFilePayload {
+  // raw data bytes for an embedded artifact
+  bytes data = 1;
+
+  // A generated staged name (no path).
```

Comment:
Should this be an actual path where the user says where in the container this file should live?

Reply:
+1 to allowing (relative) paths to some stagingDir root.

Reply:
Ack.